### ISSUE TITLE
font-size is applied when return from P to H2

### ISSUE THREAD

**Issue by steoo** [yabwe/medium-editor#840]
Scenario: 

1) Type text, markup: `<p> text </p>`
2) H2 text, resulting markup: `<h2> text </h2>`
3) Bold text, resulting markup `<h2> <b> text </b>`
3) Put the caret before the text
4) Press Return, new line is created
5) Resulting markup: `<p> <b> text </b> </p> ` ( Is it normal that the H2 is removed?)
6) Put the carert before the text and press `backspace`
7) Resulting markup: `<h2> <b font-size: calc(1.4375rem - 2px); line-height: 1.7;> text </b> </h2>`

This clearly mess up the style and the formatting

**Comment by nmielnik** [comment]
Hey @steoo would you mind adding these steps into #543 ?  I'm pretty certain this issue is another manifestation of that issue.

Also, you're saying that the `<h2>` is removed during step 5), but then the browser adds the `<h2>` back in after step 6)?  If you're able to reliably reproduce the `<h2>` getting removed, can you open a separate issue for that?

**Comment by steoo** [comment]
Of course I'll do both, only couple minutes :) !
